sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/BeamThreadsFilter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.solr;
+
+import com.carrotsearch.randomizedtesting.ThreadFilter;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Filter out Beam leak threads.
```

Comment:
This needs more documentation, from reading the code I don't understand at all what this means. Is it working around a bug in Beam, or a bug in Solr, or a bug in the testing framework, or something else?